iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -461,6 +462,25 @@ LogicalResult convertInterfaceStoreTensorOp(
 }
 
 namespace {
+
+// Folds linalg.reshape op that directly reshaping an HAL interface binding
+// subspan op into the subspan op itself.
+class FoldReshapeIntoInterfaceBindingSubspan final
```

Comment:
Can we do this folding at the Flow level? In buffer world reshapes are aliases. As we put more things into dispatch regions managing those aliases can be hard.

Reply:
Okay for me to move it earlier. Given that this patch is already fairly large, I won't do it here. I'll revert this commit and have another pull request for it.